src/sys/socket/mod.rs

```diff
@@ -1631,6 +1641,23 @@ pub fn sendmsg<S>(fd: RawFd, iov: &[IoSlice<'_>], cmsgs: &[ControlMessage],
 }
 
 
+/// Send data in scatter-gather vectors to a socket, possibly accompanied
+/// by ancillary data. Optionally direct the message at the given address,
+/// as with sendto.
+/// sendmsg_pre_alloc is the same as sendmsg but it accepts a preallocated
+/// cmsg buffer vector.
```

Comment:
Let's emphasize that `pre_alloc` part and link the original API:

```suggestion
/// `sendmsg_pre_alloc()` is the same as [`sendmsg()`] but it accepts a preallocated
/// `cmsg` buffer vector.
///
/// Send data in scatter-gather vectors to a socket, possibly accompanied
/// by ancillary data. Optionally direct the message at the given address,
/// as with sendto.
```

Reply:
btw can I rename the new API as `sendmsg_prealloc()`, I feel the underscore there is a little bit of redundant

Reply:
> btw can I rename the new API as sendmsg_prealloc()

Sure